src/scene/layer.js

```diff
@@ -734,6 +747,32 @@ class Layer {
         this.markLightsDirty();
     }
 
+    get splitLights() {
+
+        if (this._splitLightsDirty) {
+            this._splitLightsDirty = false;
+
+            const splitLights = this._splitLights;
+            for (let i = 0; i < splitLights.length; i++)
+                splitLights[i].length = 0;
+
+            const lights = this._lights;
+            for (let i = 0; i < lights.length; i++) {
+                const light = lights[i];
+                if (light.enabled) {
```

Comment:
is splitLights updated when lights are enabled and disabled?

Reply:
yes, as light is removed from the layer at that point which sets markLightsDirty to true